{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 3.7.11 error\n",
    "\n",
    "JSONDecodeError                           Traceback (most recent call last)\n",
    "~\\AppData\\Local\\Temp\\ipykernel_6480\\2266909351.py in <module>\n",
    "      3 \n",
    "      4 # 학습된 모델을 불러옵니다.\n",
    "----> 5 model = tf.keras.models.load_model(data_path / 'trained_model')\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\save.py in load_model(filepath, custom_objects, compile, options)\n",
    "    210       if isinstance(filepath, six.string_types):\n",
    "    211         loader_impl.parse_saved_model(filepath)\n",
    "--> 212         return saved_model_load.load(filepath, compile, options)\n",
    "    213 \n",
    "    214   raise IOError(\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in load(path, compile, options)\n",
    "    136   # Recreate layers and metrics using the info stored in the metadata.\n",
    "    137   keras_loader = KerasObjectLoader(metadata, object_graph_def)\n",
    "--> 138   keras_loader.load_layers()\n",
    "    139 \n",
    "    140   # Generate a dictionary of all loaded nodes.\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in load_layers(self)\n",
    "    374       self.loaded_nodes[node_metadata.node_id] = self._load_layer(\n",
    "    375           node_metadata.node_id, node_metadata.identifier,\n",
    "--> 376           node_metadata.metadata)\n",
    "    377 \n",
    "    378     for node_metadata in metric_list:\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in _load_layer(self, node_id, identifier, metadata)\n",
    "    383   def _load_layer(self, node_id, identifier, metadata):\n",
    "    384     \"\"\"Load a single layer from a SavedUserObject proto.\"\"\"\n",
    "--> 385     metadata = json_utils.decode(metadata)\n",
    "    386 \n",
    "    387     # If node was already created\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\json_utils.py in decode(json_string)\n",
    "     67 \n",
    "     68 def decode(json_string):\n",
    "---> 69   return json.loads(json_string, object_hook=_decode_helper)\n",
    "     70 \n",
    "     71 \n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\n",
    "    359     if parse_constant is not None:\n",
    "    360         kw['parse_constant'] = parse_constant\n",
    "--> 361     return cls(**kw).decode(s)\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\decoder.py in decode(self, s, _w)\n",
    "    335 \n",
    "    336         \"\"\"\n",
    "--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
    "    338         end = _w(s, end).end()\n",
    "    339         if end != len(s):\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\decoder.py in raw_decode(self, s, idx)\n",
    "    353             obj, end = self.scan_once(s, idx)\n",
    "    354         except StopIteration as err:\n",
    "--> 355             raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
    "    356         return obj, end\n",
    "\n",
    "JSONDecodeError: Expecting value: line 1 column 1 (char 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "3.9.7 error lms node와 동일한 버전\n",
    "tensorflow 2.6.0 / 2.6.0\n",
    "keras 2.6.0 /  \n",
    "numpy 1.21.4 / 1.22.2\n",
    "'''\n",
    "\n",
    "---------------------------------------------------------------------------\n",
    "JSONDecodeError                           Traceback (most recent call last)\n",
    "~\\AppData\\Local\\Temp\\ipykernel_6480\\2266909351.py in <module>\n",
    "      3 \n",
    "      4 # 학습된 모델을 불러옵니다.\n",
    "----> 5 model = tf.keras.models.load_model(data_path / 'trained_model')\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\save.py in load_model(filepath, custom_objects, compile, options)\n",
    "    210       if isinstance(filepath, six.string_types):\n",
    "    211         loader_impl.parse_saved_model(filepath)\n",
    "--> 212         return saved_model_load.load(filepath, compile, options)\n",
    "    213 \n",
    "    214   raise IOError(\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in load(path, compile, options)\n",
    "    136   # Recreate layers and metrics using the info stored in the metadata.\n",
    "    137   keras_loader = KerasObjectLoader(metadata, object_graph_def)\n",
    "--> 138   keras_loader.load_layers()\n",
    "    139 \n",
    "    140   # Generate a dictionary of all loaded nodes.\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in load_layers(self)\n",
    "    374       self.loaded_nodes[node_metadata.node_id] = self._load_layer(\n",
    "    375           node_metadata.node_id, node_metadata.identifier,\n",
    "--> 376           node_metadata.metadata)\n",
    "    377 \n",
    "    378     for node_metadata in metric_list:\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\load.py in _load_layer(self, node_id, identifier, metadata)\n",
    "    383   def _load_layer(self, node_id, identifier, metadata):\n",
    "    384     \"\"\"Load a single layer from a SavedUserObject proto.\"\"\"\n",
    "--> 385     metadata = json_utils.decode(metadata)\n",
    "    386 \n",
    "    387     # If node was already created\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\site-packages\\tensorflow\\python\\keras\\saving\\saved_model\\json_utils.py in decode(json_string)\n",
    "     67 \n",
    "     68 def decode(json_string):\n",
    "---> 69   return json.loads(json_string, object_hook=_decode_helper)\n",
    "     70 \n",
    "     71 \n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)\n",
    "    359     if parse_constant is not None:\n",
    "    360         kw['parse_constant'] = parse_constant\n",
    "--> 361     return cls(**kw).decode(s)\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\decoder.py in decode(self, s, _w)\n",
    "    335 \n",
    "    336         \"\"\"\n",
    "--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
    "    338         end = _w(s, end).end()\n",
    "    339         if end != len(s):\n",
    "\n",
    "~\\anaconda3\\envs\\3.7\\lib\\json\\decoder.py in raw_decode(self, s, idx)\n",
    "    353             obj, end = self.scan_once(s, idx)\n",
    "    354         except StopIteration as err:\n",
    "--> 355             raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
    "    356         return obj, end\n",
    "\n",
    "JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "a6d479fb901925bf95be03c57f66c8da4656e795ca75e28d88de06b246de9509"
  },
  "kernelspec": {
   "display_name": "Python 3.7.11 ('3.7')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "name": "python",
   "version": "3.7.11"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
